src/Layout.js

```diff
@@ -1068,8 +1068,32 @@ var computeLayout = (function() {
       child.nextAbsoluteChild = null;
     }
   }
+  
+  function layoutNode(node, parentMaxWidth, parentDirection) {
```

Comment:
Would you mind porting the caching implementation to be closer to the C version this way it won't diverge in subtle ways

https://github.com/facebook/css-layout/blob/master/src/Layout.c#L1205-L1235